### DISCUSSION TITLE
Any way to save txt2img Settings?

### DISCUSSION THREAD

**Discussion by lightningstrike2010** [nod-ai/SHARK#1735] [Ideas]
Curious if there is any way to save the txt2img settings and such, basically every time the ui is shutdown, everything is lost. Is there a location for the default settings we can change?

**Comment by one-lithe-rune** [comment]
Not at the moment. 

What I do is just grab the last image (or some other) I made with with txt2img and either send it there from the output gallery, or use drag and drop from the file system into the `import PNG info` box. That assumes you keep `Save prompt infomation to PNG` on, which is the default.

The only thing that doesn't pick up is any batch settings you might have.

**Comment by zune127** [comment]
How is this so difficult to add? It's basic ui design 101

**Reply by monorimet** [reply]
It's not terribly difficult but it is non-trivial to implement cleanly.

I think what this would probably look like is a "save as default" button that exports a .json / .csv to be loaded on program init. This could use the same tooling/format as the generated image metadata that @one-lithe-rune mentioned.

The metadata for each generated image keeps all the setting used for the image generation, so until now this has been the method by which users typically load in settings.

I'll submit a feature request for this so we don't lose it.

**Comment by monorimet** [comment]
I've filed a feature request here:
https://github.com/nod-ai/SHARK/issues/1843

If anyone has suggestions or specific asks about how this should look/work from a user perspective, please leave your thoughts on #1843.
